packages/wallet-ui/src/slices/walletSlice.ts

```diff
@@ -53,15 +55,19 @@ export const walletSlice = createSlice({
     ) => {
       const accountsToInsert = Array.isArray(payload) ? payload : [payload];
 
-      const accountSet = new Set(state.accounts);
+      const accountSet = new Set(
+        state.accounts.map((account) => account.address),
+      ); // Create a set of addresses
+
       for (const account of accountsToInsert) {
         if (!accountSet.has(account.address)) {
-          state.accounts.push(account.address);
+          state.accounts.push(account);
         }
       }
     },
     setCurrentAccount: (state, { payload }: { payload: Account }) => {
       state.currentAccount = payload.address;
+      state.currentAccountIndex = payload.addressIndex;
```

Comment:
```suggestion
      state.currentAccount = payload
```
nit:
i would suggest align with setAccounts

but there will be some more files (5 files) to change

but for long term this is good

as you may need more account information later (e.g account name..etc), if you keep adding new props to the component, seem is not for a good plan